soak in the translation of the "semestral project" guide (#22)
exaexa/better-mff-thesis: ch1.tex

\chapter{Introduction to thesis writing}
\label{chap:refs}

\section{So you have to do a thesis?}

% this section is translated from https://e-x-a.org/rp/ (thx to Jiri Benes)

If you are reading this, you likely have to write a thesis to finish your Bachelor or Master studies. First, congratulations on getting this far! What is a thesis though?

> Thesis is a monograph, a longer writing about a given single topic of choice. The topic is typically implied by your studies --- at the school of computer science, you will write about solving a problem with computers; at medicinal school you will write about an issue about health or anatomy, etc. To create the thesis, you typically proceed as follows:

Reviewer: Nitpick + medicinal -> medical
```suggestion
A thesis is a monograph, a longer writing about a given single topic of choice. The topic is typically implied by your studies --- at the school of computer science, you will write about solving a problem with computers; at medical school you will write about an issue about health or anatomy, etc. To create the thesis, you typically proceed as follows:
```

\item[Is my idea even relevant?]
To get the answer, go to Google Scholar and try to find articles that are not too old (say, at most 15--20 years, depending on the area), are not visibly outdated, and are relevant to your topic. If you cannot find any recent academic work on your topic, it might be too niche, already solved, or not academically interesting. Your supervisor will often help you find the latest research in the area.
\item[Is the problem I want to solve even suitable?]
No one expects you to produce a completely new and revolutionary solution to your problem. The method for solving your problem should be known and already explored --- in particular, you are not expected to make any groundbreaking scientific discoveries! You should ``just'' properly use the existing methods in a slightly novel context of your problem. For example, you may apply a known approach in a way that no one has tried before, slightly modify the approach to enable use in a seemingly different domain, or combine existing approaches in a novel way. In other words: the problem should be non-trivial, but you should know in advance how to approach the solution.
\item[Is the problem well-defined?]

> Solve one problem, and solve it well. You will often encounter a thesis reviewer telling you that your problem is not really a problem --- this is especially common for theses with topics such as ``I'll make a video game about X''. To provide an answer, you must have a direct, undeniable proof that after solving your problem is relevant and constitutes at least a minor advancement in some area (ideally, world will be much more beautiful and civilization will be better off with the problem solved). Ideally, you will be able to point out existing academic literature that states that your topic is indeed a problem worth solving.

Reviewer: ```suggestion
Solve one problem, and solve it well. You will often encounter a thesis reviewer telling you that your problem is not really a problem --- this is especially common for theses with topics such as ``I'll make a video game about X''. To provide an answer, you must have a direct, undeniable proof that your problem is relevant and its solution constitutes at least a minor advancement in some area (ideally, the world will be much more beautiful and civilization will be better off with the problem solved). Ideally, you will be able to point out existing academic literature that states that your topic is indeed a problem worth solving.
```

\item[How will I explain the problem to others?]
Write your thesis proposal as a hypothesis, and design an experiment that you can conduct to either confirm or reject this hypothesis, or to clearly determine the boundaries of its validity. Some theses are not experiments per se, but may be easily reworded as a case study, user testing, performance analysis, or theoretical proof. Mainly, the hypothesis--experiment wording helps you to convey the difficulty of the task, and clearly point out the ``unknowns'' that you aim to uncover.
\item[Am I trying to solve an exceedingly hard problem, or multiple problems at once?]

> In such cases, you might be shooting yourself in a foot: You will lose a lot of time on something that is not necessary for the thesis, and your thesis reviewer will get annoyed trying to spot the single problem that you are solving. If your thesis addresses multiple smaller problems, you must correctly frame them in a single, well-defined and topically consistent group of problems.

Reviewer: ```suggestion
In such cases, you might be shooting yourself in the foot: You will lose a lot of time on something that is not necessary for the thesis, and your thesis reviewer will get annoyed trying to spot the single problem that you are solving. If your thesis addresses multiple smaller problems, you must correctly frame them in a single, well-defined and topically consistent group of problems.
```

\item[Am I doing it right?]
To understand what is typically expected, examine some of the completed theses in your field. Look into thesis repositories from your institution\footnote{\url{https://dspace.cuni.cz/} for Charles University}, examine what makes a thesis successful, observe the different approaches to similar problems, writing styles and organizational structures. To get a good picture of the common issues and expectations, examine the official reviews from the supervisors and reviewers on several theses (these are typically deposited along with the thesis).
\item[Am I making it in time?]
Allocate a lot of time for writing, editing, and making good pictures. At the \emph{extreme barest minimum}, allocate at least one full month of work for fixing and incorporating the comments from your supervisor, especially if you never wrote a thesis before. Check the thesis submission requirements in advance; in particular, if you are supposed to print your thesis on paper, order the printing service with sufficient time reserve.
\item[Who will tell me what to do next?]

> You will. Remember that it is \emph{your} thesis --- while your supervisor will try to guide you, it is ultimately you who is responsible for your own work (and success). You are thus expected to take intiative (and many would argue that the main point of the theses is precisely to \emph{show the initiative}). In particular, never wait for your advisor to give you tasks. Be proactive, work independently, solve issues well before deadlines, come to meetings equipped with prepared questions and clearly summarized progress updates and milestones, and bring up your own ideas about next steps.

Reviewer: ```suggestion
You will. Remember that it is \emph{your} thesis --- while your supervisor will try to guide you, it is ultimately you who is responsible for your own work (and success). You are thus expected to take initiative (and many would argue that the main point of the theses is precisely to \emph{show the initiative}). In particular, never wait for your advisor to give you tasks. Be proactive, work independently, solve issues well before deadlines, come to meetings equipped with prepared questions and clearly summarized progress updates and milestones, and bring up your own ideas about next steps.
```
